ouroboros-consensus-cardano/src/unstable-cardano-testlib/Test/ThreadNet/Infra/ShelleyBasedHardFork.hs

```
@@ -382,16 +386,21 @@ instance
   Protocol info
 -------------------------------------------------------------------------------}
 
+type instance KESTracer (ShelleyBlock proto era) = KESAgentClientTrace
```

I don't see why this is necessary, if it is always the same for all Shelley Blocks and it is only used in the ShelleyBasedHardFork and the Embed.Binary modules.

---

it's only a type family so `KESAgentClientTrace` can be defined in `-protocols`, but we never do anything with a  `KESTracer blk` except pass them around -- I've changed it to just be an unconstrained type variable in `Embed.Binary`